src/explore-education-statistics-common/src/modules/charts/components/__tests__/MapBlockInternal.test.tsx

```diff
@@ -59,28 +59,28 @@ describe('MapBlockInternal', () => {
       // UK polygon
       expect(paths[0]).toHaveAttribute('fill', '#3388ff');
       // Location polygons
-      expect(paths[1]).toHaveAttribute('fill', '#5c80d6');
-      expect(paths[2]).toHaveAttribute('fill', '#86bcff');
-      expect(paths[3]).toHaveAttribute('fill', '#314573');
+      expect(paths[1]).toHaveAttribute('fill', 'rgba(145, 161, 201, 1)');
+      expect(paths[2]).toHaveAttribute('fill', 'rgba(218, 224, 237, 1)');
+      expect(paths[3]).toHaveAttribute('fill', 'rgba(71, 99, 165, 1)');
     });
 
     const legendItems = screen.getAllByTestId('mapBlock-legend-item');
 
     expect(legendItems).toHaveLength(5);
-    expect(legendItems[0]).toHaveTextContent('3.0% to 3.2%');
-    expect(legendItems[1]).toHaveTextContent('3.3% to 3.4%');
-    expect(legendItems[2]).toHaveTextContent('3.5% to 3.6%');
-    expect(legendItems[3]).toHaveTextContent('3.7% to 3.8%');
-    expect(legendItems[4]).toHaveTextContent('3.9% to 4.0%');
+    expect(legendItems[0]).toHaveTextContent('3.0% to 3.1%');
+    expect(legendItems[1]).toHaveTextContent('3.2% to 3.3%');
+    expect(legendItems[2]).toHaveTextContent('3.4% to 3.5%');
+    expect(legendItems[3]).toHaveTextContent('3.6% to 3.7%');
+    expect(legendItems[4]).toHaveTextContent('3.8% to 4.0%');
```

Comment:
Why have these changed, it doesn't look like the test data has?

Reply:
This is because the rounding algorithm changed slightly. There were some issues with the previous version as it didn't handle some edge-cases well (particularly when working with only integer data) and groups could become a little uneven in size.

We now go with an approach of trying to make all groups besides the last group the same size. Unfortunately, this last group will typically end up being smaller/larger than the other groups as a result. This can feel a little random depending on how the numbers round.